packages/perseus-editor/src/widgets/interactive-graph-editor.tsx

```diff
@@ -661,6 +662,24 @@ class InteractiveGraphEditor extends React.Component<Props> {
         // @ts-expect-error TS2739 Type 'Pick<Readonly<Props> & Readonly<{ children?: ReactNode; }>, "step" | "gridStep" | "snapStep" | "backgroundImage" | "markings" | "labels" | ... 5 more ... | "range">' is missing the following properties from type 'PerseusInteractiveGraphWidgetOptions': graph, correct
         return json;
     }
+
+    getSaveWarnings = () => {
+        const issues: Array<any | string> = [];
+
+        // A locked line on the graph cannot have length 0.
+        if (this.props.lockedFigures) {
+            _.each(this.props.lockedFigures, (figure) => {
+                if (
+                    figure.type === "line" &&
+                    lockedPointsEqual(figure.points[0], figure.points[1])
+                ) {
+                    issues.push("The line cannot have length 0.");
+                }
+            });
```

Comment:
We try to avoid using underscore if there's a native equivalent. 

```suggestion
            this.props.lockedFigures
                .filter((f) => f.type === "line")
                .forEach((figure) => {
                    if (
                        lockedPointsEqual(figure.points[0], figure.points[1])
                    ) {
                        issues.push("The line cannot have length 0.");
                    }
                });
```

Reply:
I have NO IDEA how I ended up using `_.each` 🤦🏽
I'm going to use a for of loop since that's what I would have actually done had I written this properly myself in the first place.